### PR TITLE
fix(client): assure error not called on done

### PR DESCRIPTION
**Pull request by bwanglzu**

### CODE REVIEW

File: tests/unit/clients/test_helper.py [hunk 1]
```diff
@@ -9,6 +9,17 @@
...
 
 
+class ClientTestExecutor(Executor):
+    @requests
+    def error(self, **kwargs):
+        raise NotImplementedError
+
+
+@pytest.fixture
+def flow():
```
Comment (JoanFM) [inline tests/unit/clients/test_helper.py:19]: flow_with_exception_request

**Review by JoanFM — COMMENTED** [review]
A test needs to be added where no `on_error` is passed to see that `on_done` is not called in that case

**Review by JoanFM — CHANGES_REQUESTED** [review]
Some small required changes

File: jina/clients/helper.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def callback_exec(
     """
     if on_error and response.status.code >= jina_pb2.StatusProto.ERROR:
         _safe_callback(on_error, continue_on_error, logger)(response)
-    elif on_done:
+    elif on_done and response.status.code < jina_pb2.StatusProto.ERROR:
```
Comment (deepankarm) [inline jina/clients/helper.py:82]:
```suggestion
    elif on_done and response.status.code == jina_pb2.StatusProto.SUCCESS:
```

**Review by deepankarm — CHANGES_REQUESTED** [review]

File: tests/unit/clients/test_helper.py [hunk 3]
```diff
@@ -30,3 +41,27 @@ async def test_websocket_clientlet():
...
+
+    on_error_mock = mocker.Mock()
+    on_done_mock = mocker.Mock()
+    on_always_mock = mocker.Mock()
+    with flow_with_exception_request as f:
+        f.post(
+            '', on_done=on_done_mock, on_error=on_error_mock, on_always=on_always_mock
+        )
+        on_always_mock.assert_called_once()
+        on_done_mock.assert_not_called()
```
Comment (JoanFM) [inline tests/unit/clients/test_helper.py:66]: do these asserts outside the context, otherwise if exception test may pass

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]

**Review by deepankarm — APPROVED** [review]

**Pull request merged**